test/puppetlabs/trapperkeeper/services/status/status_core_test.clj

```diff
@@ -73,13 +73,15 @@
             (is (= :unknown (:state result))))))
 
       (testing "and it is from a timeout"
-        (update-status-context status-fns "quux" "1.1.0" 1 (fn [_] (Thread/sleep 2000) {:state :running
-                                                                                        :status "aw yis"}))
+        ; Add a status function that will block indefinitely to ensure we will always
+        ; timeout
+        (update-status-context status-fns "quux" "1.1.0" 1 (fn [_] (deref (promise)) {:state :running
```

Comment:
I didn't think this change from `(Thread/sleep 2000)` to `(deref (promise))` would be necessary.